test/unit/serializer/index.spec.ts

```diff
@@ -7,13 +7,30 @@ import {
 } from '../../../src';
 import { gzip } from 'zlib';
 import { promisify } from 'util';
-import { interval } from '@codibre/fluent-iterable';
+import { constant, interval } from '@codibre/fluent-iterable';
 
+const wait = promisify(setTimeout);
 const gzipAsync = promisify(gzip);
 
 describe('index.ts', () => {
-	afterEach(() => {
-		// delete require.cache[require.resolve('../../../src/serializer/index')];
+	let protoSerialize: jest.SpyInstance;
+	let protoDeserialize: jest.SpyInstance;
+	let jsonSerialize: jest.SpyInstance;
+	let jsonDeserialize: jest.SpyInstance;
+	let gzipSerialize: jest.SpyInstance;
+	let gzipDeserialize: jest.SpyInstance;
+	let b64Serialize: jest.SpyInstance;
+	let b64Deserialize: jest.SpyInstance;
+
+	beforeEach(() => {
+		protoSerialize = jest.spyOn(ProtobufStrategy.prototype, 'serialize');
+		protoDeserialize = jest.spyOn(ProtobufStrategy.prototype, 'deserialize');
+		jsonSerialize = jest.spyOn(JsonStrategy.prototype, 'serialize');
+		jsonDeserialize = jest.spyOn(JsonStrategy.prototype, 'deserialize');
+		gzipSerialize = jest.spyOn(GzipStrategy.prototype, 'serialize');
+		gzipDeserialize = jest.spyOn(GzipStrategy.prototype, 'deserialize');
+		b64Serialize = jest.spyOn(Base64Strategy.prototype, 'serialize');
+		b64Deserialize = jest.spyOn(Base64Strategy.prototype, 'deserialize');
```

Comment:
I added a spy for each strategy operation to make sure, in all tests, that each strategy was being used